src/main/java/de/unistuttgart/bugfinder/solution/Solution.java

```diff
@@ -0,0 +1,42 @@
+package de.unistuttgart.bugfinder.solution;
+
+import de.unistuttgart.bugfinder.code.Code;
+import de.unistuttgart.bugfinder.solution.bug.Bug;
+import java.util.Set;
+import java.util.UUID;
+import javax.persistence.*;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.FieldDefaults;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class Solution {
+
+  @Id
+  @GeneratedValue(generator = "uuid")
+  UUID id;
+
+  @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+  Set<Bug> bugs;
+
+  @OneToOne
+  Code code;
+
+  public Solution(final Set<Bug> bugs) {
+    this.bugs = bugs;
+  }
+
+  public void addBug(final Bug bug) {
+    this.bugs.add(bug);
+  }
+
+  public void removeBug(final Bug bug) {
+    this.bugs.remove(bug);
+  }
```

Comment:
```suggestion
  public void addBug(final Bug... bugs) {
    this.bugs.addAll(bugs);
  }

  public void removeBugs(final Bug... bugs) {
    this.bugs.removeAll(bugs);
  }
```
I've noticed this already in another of the data classes as well, I just don't know where anymore.

Reply:
I mean…
Yes, I can somewhat understand it why it doesn't look good.
However, there might be an exception here:
If Hibernate behaves like I think it does, adding or removing multiple bugs will have serious performance implications.